distributed/deploy/adaptive.py

```diff
@@ -206,3 +205,137 @@ async def scale_up(self, n):
     @property
     def loop(self):
         return self.cluster.loop
+
+
+class ElasticAdaptive(Adaptive):
+    """Subclass of Adaptive that works better with unknown task durations."""
+
+    def __init__(
+        self,
+        cluster=None,
+        interval=None,
+        minimum=None,
+        maximum=None,
+        wait_count=None,
+        worker_key=None,
+        **kwargs,
+    ):
+        """
+        Adaptively allocate workers based on the number of unblocked tasks.
+
+        Parameters
+        ----------
+        cluster: object
+            Must have scale and scale_down methods/coroutines
+        interval : timedelta or str, default "1000 ms"
+            Milliseconds between checks
+        wait_count: int, default 3
+            Number of consecutive times that a worker should be suggested for
+            removal before we remove it.
+        worker_key: Callable[WorkerState]
+            Function to group workers together when scaling down
+            See Scheduler.workers_to_close for more information
+        minimum: int
+            Minimum number of workers to keep around
+        maximum: int
+            Maximum number of workers to keep around
+        **kwargs:
+            Extra parameters to pass to Scheduler.workers_to_close
+        """
+        super().__init__(
+            cluster=cluster,
+            interval=interval,
+            minimum=minimum,
+            maximum=maximum,
+            wait_count=wait_count,
+            target_duration=None,
+            worker_key=worker_key,
+        )
+
+    async def recommendations(self, target: int) -> dict:
+        """
+        Make scale up/down recommendations based on current state and target
+        """
```

Comment:
Why do we need another recommendation method? Can this be somehow merged (haven't investigated the diff, yet)

Reply:
The diff is that this recommendation method doesn't close workers that haven't arrived yet. 

It removes this logic (https://github.com/dask/distributed/blob/main/distributed/deploy/adaptive_core.py#L171): 
```
not_yet_arrived = requested - observed
to_close = set()
if not_yet_arrived:
to_close.update(toolz.take(len(plan) - target, not_yet_arrived))
```

This might be specific to my use case since I'm using off-prem workers that take a while to start up.